### PR TITLE
fix: removed old key_val_input, updated keyval_array, show ds info

### LINKED ISSUE #6181: [Bug]The header should not be displayed in the Active tab for the Authenticated API when there is no header
## Description

[What happened]The header should not be displayed in the Active tab for the Authenticated API when there is no header

### Steps to reproduce the behaviour:

1.Go to 'Datasource create a new tab
2.Click on Authenticated API
3.Add the URL and select Authenticate type as None don't add any header
4.Now click on save and go to the Active tab
5.Click on Show more link
6.Verify the display 
image

### Important Details

- Version: [Cloud ]
- OS: [MacOSX]
- Browser [e safari]
- Environment [Release frozen]

### LINKED ISSUE #14804: [Bug]-[6489]:Cannot delete the first header in API datasource form
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In API datasource page, headers section, the delete icon is shown only from the second row onwards. This means that I cannot delete the first row, and have to remove the contents manually.

https://user-images.githubusercontent.com/94514895/175768142-5281a866-a864-477c-b7d2-88366cb1ce56.mov

### Steps To Reproduce

1. click `+` next to `Datasources` to see the datasource pane
2. select `Authenticated API`
3. See that the first header cannot be deleted

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

This PR deals with :
- Removing the old `KeyValueInputControl` having control type `KEY_VAL_INPUT`.
- Moved form configurations having control type `KEY_VAL_INPUT` to `KEYVALUE_ARRAY`
- Updated the Array fields for `Headers`, `Query Params` etc for datasources to follow UQI based designs
- Saved values for datasources are now visible on datasource cards in active pane as well on view mode for individual datasources.

TL;DR

Fixes #14804
Fixes #6181

Screenshot 2022-12-02 at 1 11 44 PM
Screenshot 2022-12-02 at 1 11 53 PM
Screenshot 2022-12-02 at 1 11 28 PM

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by AmanAgarwal041** [comment]
Adding label `Test Plan Approved` label as specified in https://github.com/appsmithorg/appsmith/issues/14804#issuecomment-1348664134
